libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/LoginActivity.kt

```diff
@@ -675,144 +693,193 @@ open class LoginActivity : AppCompatActivity(), OAuthWebviewHelperEvents {
-        ?: qrCodeBridgeJsonRegexInternal.find(qrCodeContent)?.groups?.get(1)?.value?.let {
-            URLDecoder.decode(it, "UTF-8")
-        }
-
-    /**
-     * Creates UI Bridge API parameters from the provided JSON string.
-     * @param uiBridgeApiParameterJsonString The UI Bridge API parameters JSON
-     * string
-     * @return The UI Bridge API parameters
+     * @param qrCodeLoginUrl The QR code login URL
+     * @return Boolean true if a log in attempt is possible using the provided QR code log in URL,
+     * false otherwise
      */
-    private fun uiBridgeApiParametersFromUiBridgeApiJson(
-        uiBridgeApiParameterJsonString: String
-    ) = JSONObject(uiBridgeApiParameterJsonString).let { uiBridgeApiParameterJson ->
-        UiBridgeApiParameters(
-            uiBridgeApiParameterJson.getString(FRONTDOOR_BRIDGE_URL_KEY),
-            when (uiBridgeApiParameterJson.has(PKCE_CODE_VERIFIER_KEY)) {
-                true -> uiBridgeApiParameterJson.optString(PKCE_CODE_VERIFIER_KEY)
-                else -> null
-            }
+    fun loginWithFrontdoorBridgeUrlFromQrCode(
+        qrCodeLoginUrl: String?
+    ) = uiBridgeApiParametersFromQrCodeLoginUrl(
+        qrCodeLoginUrl
+    )?.let { uiBridgeApiParameters ->
+        loginWithFrontdoorBridgeUrl(
+            uiBridgeApiParameters.frontdoorBridgeUrl,
+            uiBridgeApiParameters.pkceCodeVerifier
         )
-    }
-
-    /**
-     * A data class representing UI Bridge API parameters provided by a login QR
-     * code.
-     */
-    private data class UiBridgeApiParameters(
-
-        /** The front door bridge URL provided by the login QR code */
-        val frontdoorBridgeUrl: String,
-
-        /** The PKCE code verifier provided by the login QR code */
-        val pkceCodeVerifier: String?
-    )
+        true
+    } ?: false
 
     // endregion
+    // region Companion
 
     companion object {
+
+        // region General Constants
+
         const val PICK_SERVER_REQUEST_CODE = 10
         private const val SETUP_REQUEST_CODE = 72
         private const val TAG = "LoginActivity"
 
-        // region QR Code Login Via UI Bridge API Constants
+        // endregion
+        // region QR Code Login Via Salesforce Identity API UI Bridge Public Implementation
+
+        /**
+         * For QR code login URLs, the URL path which distinguishes them from other URLs provided by
+         * the app's internal QR code reader or deep link intents from external QR code readers.
+         *
+         * Apps may customize this so long as it matches the server-side Apex class or other code
+         * generating the QR code.
+         *
+         * Apps need not use the QR code login URL structure provided in this companion object if
+         * they wish to entirely customize the QR code login URL format and implement a custom
+         * parsing scheme.
+         */
+        @Suppress("MemberVisibilityCanBePrivate")
+        var qrCodeLoginUrlPath = "/login/qr"
```

Comment:
Instead of private constants, I made these companion properties variables so the app could make partial adjustments to the formatting of the QR code login URL and JSON while still reusing the logic.